docs/blog/2018-2-12-case-study-mike-johnston/index.md

```diff
@@ -0,0 +1,72 @@
+
+## Obscurity isn’t Security, but it Helps
+
+The static site structure provides a layer of separation that makes potential attacks less damaging. If an attacker gains access to the machine containing the static content, they won’t discover anything that isn’t already publicly available. They can alter the content shown temporarily, but this can be solved in minutes with a new build and a new static server with fresh credentials. With some care, there should be little evidence leading back to the headless WordPress server managing content and administrator accounts.
+
+This is not an excuse for half-hearted security precautions or maintenance, however. It is still essential to keep your WordPress installation up to date and regularly audit server security. A skilled hacker often has a variety of different attack vectors they can use to gain pieces of information about your architecture — not all of them based on technology. Assume that your admin will be discovered, and take appropriate measures to keep data safe.
+
+Using tools like [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) headers, encrypted auth tokens, and secure connections you can still connect to remote APIs with strong protections for your users, systems, and data.
+
+## The Trail Ahead
+
+The WordPress solution is excellent for managing dynamic editorial content, but as we build out more interactive features we need to begin interacting with an API to show content that changes depending on the logged in user, or user-generated content. We could begin interacting with the WordPress API directly, but that works against the aforementioned security and performance benefits of using a headless instance and a static build.
+
+Instead, I decided to go with an [Apollo](https://www.apollographql.com/) GraphQL server so that I could interact with the API server the same way we were interacting with the Gatsby content system. Using standard Express middleware and Apollo tools, I was able to quickly define a schema that both my server and client code can programmatically understand and help type-check during development. It’s working quite well to help us quickly build out interfaces with minimal data issues.
+
+Security and performance will obviously become greater concerns as the interactive application is built out, but we have more time to deal with that while the homepage is already up and running and serving up dynamic content with the performance of static files.
+
+## We’re Available!
+
+While Ecliptic is finishing up the interactive features for the site, we also have wide availability opening up in February for new contracts! We can get started right away on your next project, from high-performance content-driven sites to richly interactive web and mobile applications.
```

Comment:
Please move some of this content to the beginning to provide context on your organization. Please remove the plug though as this will be an official GatsbyJS blog post. It's fine to include contact details on the bottom of the post, eg, "if you'd like to hear more about how we did this, I would love to chat!"